pkg/openapi.go

```diff
@@ -636,10 +636,22 @@ func (o *OpenAPIContext) gatherResource(
 				return errors.Errorf("%s not found in api schemas for discriminated type in path %s", schemaName, apiPath)
 			}
 
+			var resourceTypeToken *string
+			var err error
 			// Don't prefix the parent name since this resource
 			// will already be scoped under a module.
 			discriminatedResourceName := ToPascalCase(discriminatedValue)
-			resourceTypeToken, err := o.gatherResourceProperties(discriminatedResourceName, *typeSchema.Value, resourceResponseType, apiPath, module)
+			if resourceResponseType != nil && resourceResponseType.Discriminator != nil {
+				responseSchemaRef := resourceResponseType.Discriminator.Mapping[discriminatedValue]
```

Comment:
The discriminated values will be the same between the request and the response since the API can't possibly return a new type that it can't handle a request for. In other words, consider the following discriminator mapping in the request schema:

```
discriminator:
  propertyName: type
  mapping:
    "cat": "#/components/schemas/Cat"
```

The API can't return a dog type in the response, which wouldn't make sense.